PresentScreenings/Utilities/ScreeningsPlanner.cs

```diff
@@ -115,6 +116,103 @@ public static string LogTimeString()
         #endregion
 
         #region Private Methods
+        private void AttendScreenings(string filmFan, List<Screening> screenings)
+        {
+            DateTime orgDay = _controller.Plan.CurrDay;
+            foreach (var screening in screenings)
+            {
+                bool fits = true;
+                if (screening is OnDemandScreening onDemandScreening)
+                {
+                    fits = FitOnDemandScreening(filmFan, onDemandScreening);
+                }
+                if (fits && screening.IsPlannable)
+                {
+                    screening.ToggleFilmFanAttendance(filmFan);
+                    screening.AutomaticallyPlanned = true;
+                    _controller.UpdateAttendanceStatus(screening);
+                    _plannedScreenings.Add(screening);
+                }
+                _controller.ReloadScreeningsView();
+            }
+            _controller.GoToDay(orgDay);
+        }
+
+        private bool FitOnDemandScreening(string filmFan, OnDemandScreening onDemandScreening)
+        {
+            // Define helper functions.
+            bool fits(OnDemandScreening odScreening) => odScreening.IsPlannable
+                && !odScreening.HasTimeOverlap
+                && odScreening.FitsAvailability;
+            bool canMove(OnDemandScreening odScreening) =>
+                odScreening.AttendingFilmFans.Count == 0;
+
+
+            // No moving if the screening already fits.
+            if (fits(onDemandScreening))
+            {
+                return true;
+            }
+
+            // Try to fit the screening by moving it.
+            bool stop = false;
+            bool found = false;
+            bool tryNextDay = false;
+            if (canMove(onDemandScreening))
+            {
+                // Start at the first allowed start time of the screening.
+                _controller.MoveToWindowStart(onDemandScreening);
+                if (fits(onDemandScreening))
+                {
+                    return true;
+                }
+
+                // Try to fit moving forward.
+                while (!stop)
+                {
+                    TimeSpan span = _controller.GetSpanToAutomaticallyFit(onDemandScreening);
+                    if (span == TimeSpan.Zero)
+                    {
+                        tryNextDay = true;
+                    }
+                    else
+                    {
+                        DateTime orgStartTime = onDemandScreening.StartTime;
+                        _controller.MoveOnDemandScreeningAutomatically(onDemandScreening, span);
+                        if (onDemandScreening.StartTime == orgStartTime || onDemandScreening.StartTime == onDemandScreening.WindowEndTime)
```

Comment:
Please remove the second term, this never occurs in automatic planning.
Start time is move to maximal one minute before the window end time.

Reply:
The first term is also always `false` because the `if` statement guarantees that the time span to move is not `TimeSpan.Zero`.
So I removed the whole condition.